File: OP_lab/lab5/c#/letter_string.cs
namespace lab_5_OP
{
    public class LetterString : MyString
    {
        public LetterString() : base(){}
        public LetterString(char[] ch) : base(ch){}
        public LetterString(LetterString other) : base(other){}

        public void RowOffset()
        {
            if (this._str.Length == 0) return;
            var tmp = this._str[this.GetSize() - 1];
            for (int i = this.GetSize()-1; i > 0; --i)
            {
                _str[i] = _str[i - 1];
            }

            _str[0] = tmp;
        }

        public char[] get_string()
        {
            char[] tmp = new char[this.GetSize()];
            for (int i = 0; i < this.GetSize(); i++)
            {
                tmp[i] = this._str[i];
            }

            return tmp;
        }
    }
}

File: OP_lab/lab5/c++/letter_string.cpp
//
// Created by maxim on 24.03.2020.
//

#include "letter_string.h"

void letter_string::row_offset() {
    if(this->size == 0) return;
    char tmp = this->str[size-1];
    for (int i = this->size - 1; i > 0; --i) {
        str[i] = str[i-1];
    }
    str[0] = tmp;
}

char *letter_string::get_string() {
    auto* tmp = new char[this->size];
    for (int i = 0; i < this->size; ++i) {
        tmp[i] = this->str[i];
    }
    return tmp;
}
